lib/cmxl/fields/transaction.rb

```diff
@@ -19,7 +19,27 @@ def credit?
       end
 
       def debit?
-        !credit?
+        self.data['funds_code'].to_s.upcase == 'D'
+      end
+
+      def storno_credit?
+        credit? && storno?
+      end
+
+      def storno_debit?
+        debit? && storno?
+      end
+
+      def storno?
+        !storno_flag.empty?
+      end
+
+      def funds_code
+        [storno_flag, super].join
```

Comment:
why `super` here and not `data['funds_code']` - bei `debit?/credit?` verwenden wir auch den data hash.

Reply:
from `Field`, which `Transaction` extends:

```ruby
def method_missing(m, *value)
  if m =~ /=\z/
    self.data[m] = value.first
  else
    self.data[m.to_s]
  end
end
```
so that's basically what happens here. 

`debit?` and `credit?` don't have a super. (or it would return nothing). we could call `debit` or `credit` instead of `data[something]` though. good point

Reply:
yes, `self.funds_code` could be used instead of the `self.data['funds_code']` but I don't know if that is nicer or any helpful.